src/BTCPayServer.Lightning.LNDhub/LndHubClient.cs

```diff
@@ -32,7 +33,11 @@ public class LndHubClient
         private static readonly HttpClient _sharedClient = new ();
         private static readonly ConcurrentDictionary<string, AuthResponse> _cache = new();
         public readonly string CacheKey;
-        private static readonly AsyncDuplicateLock _locker = new();
+        private static readonly AsyncKeyedLocker<string> _locker = new(o =>
+        {
+            o.PoolSize = 20;
```

Comment:
Why do we need these settings? Does this restrict to only 20 concurrent locks?

Reply:
No, the pooling is there to improve performance and reduce memory allocations by reusing objects. In case the pool is empty, an object is created. The pool is thus not a restriction.